test/e2e/framework/test_context.go

```diff
@@ -172,3 +175,29 @@ func RegisterNodeFlags() {
 	flag.StringVar(&TestContext.EvictionHard, "eviction-hard", "memory.available<250Mi,imagefs.available<10%%", "The hard eviction thresholds. If set, pods get evicted when the specified resources drop below the thresholds.")
 	flag.StringVar(&TestContext.ManifestPath, "manifest-path", "", "The path to the static pod manifest file.")
 }
+
+// Enable viper configuration management of flags.
+func ViperizeFlags() {
+	// Add viper in a minimal way.
+	// Flag interop isnt possible, since 'go test' coupling to flag.Parse.
+	viper.SetConfigName("e2e")
+	viper.AddConfigPath(".")
+	viper.ReadInConfig()
+
+	// TODO @jayunit100: Maybe a more elegant viper-flag integration for the future?
+	// For now, we layer it on top, because 'flag' deps of 'go test' make pflag wrappers
+	// fragile, seeming to force 'flag' to have deep awareness of pflag params.
+	RegisterCommonFlags()
+	RegisterClusterFlags()
+	flag.Parse()
+
+	viperFlagSetter := func(f *flag.Flag) {
+		if viper.IsSet(f.Name) {
+			fmt.Println("[overwrite] viper found a Settting for  ", f.Name, " ", f.Value)
```

Comment:
Do we want to print this all out, seems more like debug only level printing.   We have so much spam today, and I think only the sig-testing folks care about this.